compiler/src/main/java/com/xfinity/resourceprovider/RpCodeGenerator.java

```diff
@@ -78,6 +87,10 @@ TypeSpec generateResourceProviderClass() {
         FieldSpec integerProvider = FieldSpec.builder(integerProviderClassName, "integerProvider")
                                              .addModifiers(Modifier.PRIVATE, Modifier.FINAL).build();
 
+        ClassName idProviderClassName = get(packageName, "IdProvider");
```

Comment:
Non-blocking, since this is fiddly - should this also be scoped to only get made if the user requested Id provision?